app/routes/participate/survey.js

```diff
@@ -6,10 +6,21 @@ import config from 'ember-get-config';
 export default Ember.Route.extend(AuthenticatedRouteMixin, {
     _experiment: null,
     _session: null,
-    store: Ember.inject.service(),
+
     currentUser: Ember.inject.service(),
     i18n: Ember.inject.service(),
+    session: Ember.inject.service(),
 
+    init() {
+        // If any request fails due to authentication reasons, kick user to login page
+        // Selectively adapted from:
+        //  https://github.com/simplabs/ember-simple-auth/blob/1.2.0-beta.1/addon/mixins/application-route-mixin.js#L113
+        this.get('session').on('invalidationSucceeded', Ember.run.bind(this, () => {
+            // Since the user has work in progress, we must turn off the prompt-on-exit behavior.
+            Ember.$(window).off('beforeunload');
+            window.location.replace('');
```

Comment:
Good questions!

Two main reasons:
1. We don't want to reload and be back at the survey page; we consciously want to kick the user back to the landing URL (login)
2. Parity with Ember-simple-auth. Although this app's requirements are a bit too weird to use the functionality directly, this fix largely parallels what the auth framework would normally try to do for free.  https://github.com/simplabs/ember-simple-auth/blob/1.2.0-beta.1/addon/mixins/application-route-mixin.js#L133